pkg/webhook/server/certificate/manager.go

```diff
@@ -0,0 +1,166 @@
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+type manager struct {
+	crMgr       crmanager.Manager
+	certManager certificate.Manager
+	certStore   *filePairStore
+	log         logr.Logger
+}
+
+type WebhookType string
+
+const (
+	MutatingWebhook   WebhookType = "Mutating"
+	ValidatingWebhook WebhookType = "Validating"
+)
+
+// NewManager with create a certManager that generated a pair of files ${certDir}/${certFile} and ${certDir}/${keyFile} to use them
+// at the webhook TLS http server.
+// It will also starts at cert manager [1] that will update them if they expire.
+// The generate certificate include the following fields:
+// DNSNames (for every service the webhook refers too):
+//	   - ${service.Name}
+//	   - ${service.Name}.${service.namespace}
+//	   - ${service.Name}.${service.namespace}.svc
+// Subject:
+// 	  - CN: ${webhookName}
+// Usages:
+//	   - UsageDigitalSignature
+//	   - UsageKeyEncipherment
+//	   - UsageServerAuth
+//
+// It will also update the webhook caBundle field with the cluster CA cert and
+// approve the generated cert/key with k8s certification approval mechanism
+func NewManager(
+	crMgr crmanager.Manager,
+	webhookName string,
+	webhookType WebhookType,
+	certDir string, certFileName string, keyFileName string) (*manager, error) {
+
+	certStore, err := NewFilePairStore(
+		certDir,
+		certFileName,
+		keyFileName,
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to initialize new webhook cert/key file store")
+	}
+	m := &manager{
+		crMgr:     crMgr,
+		log:       logf.Log.WithName("webhook/server/certificate/manager"),
+		certStore: certStore,
+	}
+
+	dnsNames := []string{}
+	if webhookType == MutatingWebhook {
+		mutatingWebHookConfig, err := m.updateMutatingWebhookCABundle(webhookName)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to update CA bundle at webhook")
+		}
+
+		for _, webhook := range mutatingWebHookConfig.Webhooks {
+			dnsNames = append(dnsNames, dnsNamesForService(*webhook.ClientConfig.Service)...)
+		}
+	} else if webhookType == ValidatingWebhook {
+		validatingWebHookConfig, err := m.updateValidatingWebhookCABundle(webhookName)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to update CA bundle at webhook")
+		}
+
+		for _, webhook := range validatingWebHookConfig.Webhooks {
+			dnsNames = append(dnsNames, dnsNamesForService(*webhook.ClientConfig.Service)...)
+		}
+	}
+	certConfig := certificate.Config{
+		ClientFn: func(current *tls.Certificate) (certificatesclientv1beta1.CertificateSigningRequestInterface, error) {
+			certClient, err := certificatesclientv1beta1.NewForConfig(crMgr.GetConfig())
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to create cert client for webhook")
+			}
+			return newCSRApprover(certClient.CertificateSigningRequests()), nil
+		},
+		Template: &x509.CertificateRequest{
+			Subject: pkix.Name{
+				CommonName: fmt.Sprintf("%s", webhookName),
```

Comment:
webhookName is already string, you don't need to format it

Reply:
It was something else before, now is just the hook name.